src/__tests__/getLocationForJsonPath.spec.ts

```diff
@@ -94,4 +96,76 @@ describe('getLocationForJsonPath', () => {
       });
     });
   });
+
+  describe('CRLF fixture', () => {
+    const resultCRLF = parseWithPointers(spectralCRLF);
+    const resultLF = parseWithPointers(spectralLF);
+
+    test.each`
+      start     | end        | path                   | closest
+      ${[0, 0]} | ${[5, 9]}  | ${['servers']}         | ${true}
+      ${[2, 5]} | ${[4, 13]} | ${['info']}            | ${true}
+      ${[2, 5]} | ${[4, 13]} | ${['info', 'server']}  | ${true}
+      ${[2, 5]} | ${[4, 13]} | ${['info', 'contact']} | ${true}
+      ${[4, 9]} | ${[4, 13]} | ${['info', 'title']}   | ${true}
+      ${[5, 6]} | ${[5, 9]}  | ${['paths']}           | ${true}
+    `('should return proper location for given JSONPath $path', ({ start, end, path, closest }) => {
+      expect(getLocationForJsonPath(resultCRLF, path, closest)).toEqual(
+        getLocationForJsonPath(resultLF, path, closest)
+      );
+      expect(getLocationForJsonPath(resultCRLF, path)).toEqual(getLocationForJsonPath(resultLF, path));
+      expect(getLocationForJsonPath(resultCRLF, path, closest)).toEqual(
+        start.length > 0 && end.length > 0
+          ? {
+              range: {
+                start: {
+                  character: start[1],
+                  line: start[0],
+                },
+                end: {
+                  character: end[1],
+                  line: end[0],
+                },
+              },
+            }
+          : void 0
+      );
+    });
+  });
+
+  describe('shifted fixture', () => {
+    const result = parseWithPointers(`--- foobar: bar`);
+
+    test.each`
+      start     | end        | path         | closest
+      ${[0, 4]} | ${[0, 15]} | ${['paths']} | ${true}
+    `('should return proper location for given JSONPath $path', ({ start, end, path, closest }) => {
+      expect(getLocationForJsonPath(result, path, closest)).toEqual(
+        start.length > 0 && end.length > 0
+          ? {
+              range: {
+                start: {
+                  character: start[1],
+                  line: start[0],
+                },
+                end: {
+                  character: end[1],
+                  line: end[0],
+                },
+              },
+            }
+          : void 0
+      );
+    });
+  });
+
+  it('should handle null-ish items', () => {
+    const result = parseWithPointers(`----~
```

Comment:
Discovered this issue while playing around with fixtures.
Unfortunately, yaml-ast-parser typings are broken, and they don't take nulls into account at all, even though most of the nodes or properties are nullable.